src/test/java/test/DynamicGraphTest.java

```diff
@@ -74,15 +73,19 @@ public void test8() {
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
     dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
-
-    assertFreeNodesEquals(dg, a1, a2);
+    assertFreeNodesEquals(dg);
+    
+    dg.setStatus(x, Status.FINISHED);
+    dg.setStatus(y, Status.FINISHED);
+    assertFreeNodesEquals(dg,a1);
+    
     dg.setStatus(dg.getFreeNodes(), Status.RUNNING);
 
     dg.setStatus(a1, Status.FINISHED);
     assertFreeNodesEquals(dg);
 
     dg.setStatus(a2, Status.FINISHED);
-    assertFreeNodesEquals(dg, b1, b2);
+    assertFreeNodesEquals(dg, b1);
 
     dg.setStatus(b2, Status.RUNNING);
```

Comment:
A circular dependency is forbidden when it involves 2 edges with the same weight.

But
```java
public class CircularDependencies {
	@Test(priority=1, dependsOnMethods="testB")
	public void testA(){
	}	
	@Test(priority=2)
	public void testB(){
	}
}
```
should work even if there is a circular dependency (testA -> testB, with dependsOnMethods, and testB -> testA, with priority).
But as `dependsOnMethods` is more important than `priority` then `testB` should be the first run method.